tools/convert_posts_to_page_bundles.py

```
@@ -0,0 +1,67 @@
+def download_image(url, output_path):
+    try:
+        response = requests.get(url)
+        response.raise_for_status()
+        with open(output_path, 'wb') as f:
+            f.write(response.content)
+        return True
+    except requests.RequestException as e:
+        logging.error(f"Failed to download image from {url}: {e}")
+        return False
+
+def process_images(content, post_dir):
```

Ruff (0.8.2)

Missing return type annotation for public function `process_images`

(ANN201)